homeassistant/components/anova/__init__.py

```diff
@@ -73,6 +59,24 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
-        hass.data[DOMAIN].pop(entry.entry_id)
-
+        anova_data: AnovaData = hass.data[DOMAIN].pop(entry.entry_id)
+        # Disconnect from WS
+        await anova_data.api.disconnect_websocket()
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+
+    if config_entry.version == 1:
+        # It used to be needed to persist devices, however that is no longer the case as the
+        # websocket holds information for all devices on the account.
+        new = {
+            CONF_USERNAME: config_entry.data[CONF_USERNAME],
+            CONF_PASSWORD: config_entry.data[CONF_PASSWORD],
+        }
+
+        config_entry.version = 2
+        hass.config_entries.async_update_entry(config_entry, data=new)
+
+    return True
```

Comment:
Doing a migration makes the change backwards incompatible, and since we don't need to restructure data, only ignore, I would maybe just suggest to keep rolling with the current config entry data

Reply:
Let me recheck this with others

Reply:
Ever hear anything about this? Should I avoid doing a migration?

Reply:
Please don't do the migration if it's only about cleaning up unneeded data.
Instead, please add a comment saying this cleanup can happen 6 months from now, with a migration to config entry 1.2

Reply:
Removed and set back to 1